src/DataTypes/EnumValues.cpp

```diff
@@ -52,7 +55,23 @@ void EnumValues<T>::fillMaps()
 template <typename T>
 T EnumValues<T>::getValue(StringRef field_name, bool try_treat_as_id) const
 {
-    const auto it = name_to_value_map.find(field_name);
+    ContextPtr query_context;
+    if (CurrentThread::isInitialized())
+    {
+        query_context = CurrentThread::get().getGlobalContext();
```

Comment:
This looks strange - it is better to push it to FormatSettings and drag through function arguments here.